go/util/resources.go

```diff
@@ -0,0 +1,108 @@
+package util
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	// "github.com/davecgh/go-spew/spew"
+	log "github.com/skaes/logjam-tools/go/logging"
+)
+
+type Resources struct {
+	TimeResources        []string `json:"time_resources"`
+	CallResources        []string `json:"call_resources"`
+	MemoryResources      []string `json:"memory_resources"`
+	HeapResources        []string `json:"heap_resources"`
+	TimeAndCallResources []string
+}
+
+func (r *Resources) Initialize() {
+	m, n := len(r.TimeResources), len(r.CallResources)
+	r.TimeAndCallResources = make([]string, m+n, m+n)
+	for i, resource := range r.TimeResources {
+		r.TimeAndCallResources[i] = resource
+	}
+	for i, resource := range r.CallResources {
+		r.TimeAndCallResources[i+m] = resource
+	}
+}
+
+func RetrieveResources(url, env string) *Resources {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		log.Error("could not create http request: %s", err)
+		return nil
+	}
+	req.Header.Add("Accept", "application/json")
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		log.Error("could not retrieve resources: %s", err)
+		return nil
+	}
+	if res.StatusCode != 200 {
+		log.Error("unexpected response: %d", res.Status)
+		ioutil.ReadAll(res.Body)
```

Comment:
This reads the body but does not close it. We are currently only deferring this in line 54 (after the return). I'd move it up.

Reply:
damn.